support/datastore/ledgerbatch_config_test.go

```diff
@@ -35,3 +37,41 @@ func TestGetObjectKeyFromSequenceNumber(t *testing.T) {
 		})
 	}
 }
+
+func TestGetObjectKeyFromSequenceNumber_ObjectKeyDescOrder(t *testing.T) {
+	config := LedgerBatchConfig{
+		LedgersPerFile:    1,
+		FilesPerPartition: 10,
+		FileSuffix:        ".xdr.gz",
+	}
+	sequenceCount := 10000
+	sequenceMap := make(map[uint32]string)
+	keys := make([]uint32, len(sequenceMap))
+	count := 0
+
+	for {
+		if count >= sequenceCount {
+			break
+		}
+		randSequence := rand.Uint32()
+		if _, ok := sequenceMap[randSequence]; ok {
+			continue
+		}
+		sequenceMap[randSequence] = config.GetObjectKeyFromSequenceNumber(randSequence)
+		keys = append(keys, randSequence)
+		count++
+	}
+
+	sort.Slice(keys, func(i, j int) bool {
+		return keys[i] < keys[j]
+	})
+
+	prev := sequenceMap[keys[0]]
+	for i := 1; i < sequenceCount; i++ {
+		curr := sequenceMap[keys[i]]
+		if prev < curr {
```

Comment:
```suggestion
		if prev <= curr {
```

on the off chance that the random generator returns the same value more than once